src/main/java/cubicchunks/server/PlayerCubeMapEntry.java

```diff
@@ -123,22 +125,18 @@ public void removePlayer(EntityPlayerMP player) {
 	}
 
 	public boolean providePlayerCube(boolean canGenerate) {
-		if (this.cube != null) {
-			return true;
-		}
 		int cubeX = getX(cubeAddress);
 		int cubeY = getY(cubeAddress);
 		int cubeZ = getZ(cubeAddress);
 
-		playerCubeMap.getWorld().getProfiler().startSection("loadCube");
+		playerCubeMap.getWorld().getProfiler().startSection("getCube");
 		if (canGenerate) {
-			this.cubeCache.loadCube(cubeX, cubeY, cubeZ, LOAD_OR_GENERATE);
+			this.cube = this.cubeCache.getCube(new CubeCoords(cubeX, cubeY, cubeZ), IProviderExtras.Requirement.LIGHT);
```

Comment:
I'm also not happy with wrapping the whole thing in coords (See my first review, heap/memory churn, etc)